test/unit/ResourceTypeRegistry.js

```diff
@@ -24,11 +24,29 @@ describe("ResourceTypeRegistry", function() {
     registry = new ResourceTypeRegistry();
   });
 
+  describe("constructor", () => {
+    it("should register resource descriptions provided in first parameter", () => {
+      registry = new ResourceTypeRegistry([{
+        type: "someType",
+        info: "provided to constructor"
+      }]);
+      expect(registry.type("someType")).to.be.an.object;
+      expect(registry.type("someType").info).to.equal("provided to constructor");
+    });
+
+    it("should merge global defaults into the second paramter and save as _resourceDefaults property", () => {
```

Comment:
I'm not sure we should have a test for this, especially if we add the assertion I mentioned below, since `_resourceDefaults` presumably isn't part of the public API, and I'm not sure we want to signal otherwise. Also, should `_resourceDefaults` be renamed to `_descriptionDefaults` in `ResourceTypeRegistry`? That's the name you gave it in the test below, and it's actually a better one imo. I know we already have an instance variable called `_resourceTypes`, so I'm guessing that the reasoning behind `_resourceDefaults` was to match that. But `_resourceTypes` is a type name => description map, so I think it's fine to call the other variable `_descriptionDefaults`.